src/uu/pidwait/src/pidwait.rs

```diff
@@ -0,0 +1,246 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use clap::{arg, crate_version, Arg, ArgAction, ArgMatches, Command};
+use regex::Regex;
+use std::{collections::HashSet, env, sync::OnceLock};
+use uu_pgrep::process::{walk_process, ProcessInformation, RunState, Teletype};
+use uucore::{
+    error::{UResult, USimpleError},
+    format_usage, help_about, help_usage,
+};
+use wait::waiting;
+
+mod wait;
+
+const ABOUT: &str = help_about!("pidwait.md");
+const USAGE: &str = help_usage!("pidwait.md");
+
+static REGEX: OnceLock<Regex> = OnceLock::new();
+
+#[derive(Debug)]
+struct Settings {
+    echo: bool,
+    count: bool,
+    full: bool,
+    ignore_case: bool,
+    newest: bool,
+    oldest: bool,
+    older: Option<u64>,
+    terminal: Option<HashSet<Teletype>>,
+    exact: bool,
+    runstates: Option<RunState>,
+}
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+
+    let settings = Settings {
+        echo: matches.get_flag("echo"),
+        count: matches.get_flag("count"),
+        full: matches.get_flag("full"),
+        ignore_case: matches.get_flag("ignore-case"),
+        newest: matches.get_flag("newest"),
+        oldest: matches.get_flag("oldest"),
+        older: matches.get_one::<u64>("older").copied(),
+        terminal: matches
+            .get_many::<Teletype>("terminal")
+            .map(|it| it.cloned().collect()),
+        exact: matches.get_flag("exact"),
+        runstates: matches.get_one::<RunState>("runstates").cloned(),
+    };
+
+    let pattern = pattern_initialize(&matches, &settings)?;
+    REGEX
+        .set(Regex::new(&pattern).map_err(|e| USimpleError::new(2, e.to_string()))?)
+        .unwrap();
+
+    if (!settings.newest
+        && !settings.oldest
+        && settings.runstates.is_none()
+        && settings.older.is_none()
+        && settings.terminal.is_none())
+        && pattern.is_empty()
+    {
+        return Err(USimpleError::new(
+            2,
+            "no matching criteria specified\nTry `pidwait --help' for more information.",
+        ));
+    }
+
+    let mut proc_infos = collect_proc_infos(&settings);
+
+    // For empty result
+    if proc_infos.is_empty() {
+        uucore::error::set_exit_code(1);
+    }
+
+    // Process outputs
+    if settings.count {
+        println!("{}", proc_infos.len())
+    }
+
+    if settings.echo {
+        for ele in proc_infos.iter_mut() {
+            println!("waiting for {} (pid {})", ele.status()["Name"], ele.pid)
```

Comment:
With `--newest` and `--oldest` no name is output:
```
$ pidwait --echo --newest
waiting for  (pid 35582)
```

Reply:
Fixed.

```
❯ cargo run -q pidwait -e -n
waiting for  (pid 214306)
```